### ISSUE TITLE
Unable to parse `proc {|a, *b, c, d|}`

### ISSUE THREAD

**Issue by vais** [opal/opal#1315]
Blocks Opal from being able to run `rubyspec/language/block_spec.rb`
```
$ bundle exec rake mspec_rubyspec_nodejs PATTERN=spec/rubyspec/language/block_spec.rb

...

/Users/vais/opal/lib/opal/parser.rb:82:in `on_error': An error occurred while compiling: rubyspec/language/block_spec (RuntimeError)

parse error on value "c" (tIDENTIFIER) :rubyspec/language/block_spec:32
Source: rubyspec/language/block_spec:32:27
      m([1, 2]) { |a, *b, c, d| [a, b, c, d] }.should == [1, [], 2, nil]
~~~~~~~~~~~~~~~~~~~~~~~~~~^
	from /Users/vais/.rbenv/versions/2.2.3/lib/ruby/2.2.0/racc/parser.rb:258:in `_racc_do_parse_c'
	from /Users/vais/.rbenv/versions/2.2.3/lib/ruby/2.2.0/racc/parser.rb:258:in `do_parse'
	from /Users/vais/opal/lib/opal/parser.rb:54:in `parse_to_sexp'
	from /Users/vais/opal/lib/opal/parser.rb:39:in `parse'
	from /Users/vais/opal/lib/opal/compiler.rb:149:in `compile'
	from /Users/vais/opal/lib/opal/builder_processors.rb:91:in `compiled'
	from /Users/vais/opal/lib/opal/builder_processors.rb:101:in `requires'
	from /Users/vais/opal/lib/opal/builder.rb:139:in `process_require'
	from /Users/vais/opal/lib/opal/builder.rb:42:in `block in build_str'
	from /Users/vais/opal/lib/opal/builder.rb:42:in `map'
	from /Users/vais/opal/lib/opal/builder.rb:42:in `build_str'
	from /Users/vais/opal/lib/opal/cli.rb:111:in `block in build'
	from /Users/vais/opal/lib/opal/cli.rb:159:in `evals_or_file'
	from /Users/vais/opal/lib/opal/cli.rb:110:in `build'
	from /Users/vais/opal/lib/opal/cli.rb:121:in `compiled_source'
	from /Users/vais/opal/lib/opal/cli.rb:85:in `run_code'
	from /Users/vais/opal/lib/opal/cli.rb:71:in `run'
	from bin/opal:22:in `<main>'
```

**Comment by iliabylich** [comment]
I guess parsing issue can be closed due to https://github.com/opal/opal/pull/1358 - sexp is correct, but there's an issue with `Opal::Nodes::NodeWithArgs#compile_rest_arg` - it's not 100% correct

**Comment by vais** [comment]
@iliabylich :clap:

FYI, my purpose for submitting this series of issues was basically to document the remaining issues that are blocking Opal from *running* (which does not necessarily mean *passing*) the entire Ruby Spec Suite (the subset of it that we care about anyway). So, when you resolve one of these blockers, but it happens to "unlock" a new blocker, please mark the original issue as ok to close and submit a new issue for the new blocker. Or update the existing issue with the information about the new blocker. I think closing and opening a new issue is better... cc @elia

**Comment by iliabylich** [comment]
Fixed in 31f3008f9326af85b0353dac08d9d81c73635bf6
